pnpm/src/errorHandler.ts

```diff
@@ -1,7 +1,30 @@
+import { promisify } from 'util'
 import { logger } from '@pnpm/logger'
+import pidTree from 'pidtree'
 import { REPORTER_INITIALIZED } from './main'
 
-export function errorHandler (error: Error & { code?: string }) {
+const getDescendentProcesses = promisify((pid: number, callback: (error: Error | undefined, result: number[]) => void) => {
+  pidTree(pid, { root: false }, callback)
+})
+
+const killProcesses = async () => {
```

Comment:
is this doing the same as [tree-kill](https://www.npmjs.com/package/tree-kill)? We already have tree-kill in the dependencies.

Reply:
Yes, at first I used tree-kill directly, but it seems to have changed the original  process status to `null`,  Many of our test cases rely on a status of 1 after the process exits, so I added this dependency specifically to clean up the child processes

Reply:
I have tried to replace it with kill-tree and your test is still passing.

Reply:
Did you test it on windows, the[ CI of this commit ](https://github.com/pnpm/pnpm/pull/6190/checks?check_run_id=11816473188)is reporting errors on linux, I get similar errors on my local macOS.

Reply:
Looks like it passes on Linux and Windows: https://github.com/pnpm/pnpm/actions/runs/4388916128/jobs/7685941878

and locally on my mac

Reply:
Actually on Linux it did not fail but Jest failed with this error: This usually means that there are asynchronous operations that weren't stopped in your tests. Consider running Jest with `--detectOpenHandles` to troubleshoot this issue.

So this might be that.